app.js

```diff
@@ -1,14 +1,46 @@
 const express = require("express");
 const app = express();
-const { getTopics, getApi } = require("./controllers/get.api.controllers");
+const {
+  getTopics,
+  getApi,
+  getArticleById,
+} = require("./controllers/get.api.controllers");
 
 app.use(express.json());
 
 app.get("/api/topics", getTopics);
 app.get("/api", getApi);
+app.get("/api/articles/:article_id", getArticleById);
 
-app.use((req, response) => {
+//error handling middleware
+app.use((req, response, next) => {
   response.status(404).send({ msg: "not found" });
+  next(err);
+});
+
+app.use((err, req, res, next) => {
+  if (err.status === 404 && err.msg) {
+    res.status(404).send(err);
+  }
+  next(err);
+});
+
+app.use((err, req, res, next) => {
+  if (err.status === 400 && err.msg) {
+    res.status(400).send(err);
+  }
+  next(err);
+});
```

Comment:
I would combine these two .uses

Since they're testing for pretty much the same thing - a status and a message, you can combine these two into once statement.
```

if ( err.status && err.msg) {
    // send the err.status and the err.msg
}

```
This covers both cases, and makes your code cleaner!